<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
       xmlns:context="http://www.springframework.org/schema/context"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="http://www.springframework.org/schema/beans
           http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
           http://www.springframework.org/schema/context
           http://www.springframework.org/schema/context/spring-context.xsd">

 <context:component-scan base-package="aop"></context:component-scan>
 
 <bean id="calcProxy" class="org.springframework.aop.framework.ProxyFactoryBean">
  <!-- ref에 calc를 주었기 때문에 예외 발생(calculatorImpl로 주던가) CalculatorImpl에 Component 애노테이션에 name을 설정 -->
  <!-- <property name="target" ref="calc"></property> -->
  <property name="target" ref="calculatorImpl"></property>
  <property name="proxyInterfaces">
   <list>
    <value>aop.Calculator</value>
   </list>
  </property> 
  <property name="interceptorNames">
   <list>
     <value>logPrintAroundAdvice</value><!-- 전후 Advice  -->
     <value>logPrintBeforeAdvice</value><!-- 전 Advice -->
     <value>logPrintAfterReturningAdvice</value><!-- 후 Advice -->
   </list>
  </property>
 </bean>

</beans>